worker/caasunitprovisioner/application_worker.go

```diff
@@ -175,6 +175,12 @@ func (aw *applicationWorker) loop() error {
 			if err != nil && !errors.IsNotFound(err) {
 				return errors.Trace(err)
 			}
+			// update svc info (addresses etc.) cloudservices.
+			if err = updateApplicationService(
```

Comment:
This shouldn't be done here - we call clusterChanged) below.
Doing it here means we are making 2 API calls when we only need one